core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -0,0 +1,253 @@
+                                              oracleExportCommand, config.oracleServerHostname, exitCode));
+        }
+        String out = CharStreams.toString(outBuffer);
+        //SQLPLUS command errors are not fetched from session.getStderr().
+        //Errors and output received after executing the command in SQLPlus prompt are the one that
+        //are printed on the SQL prompt
+        if (out.contains("ERROR at line")) {
+          throw new IOException(String.format("Error executing sqlplus query %s on hostname %s; error message: %s",
+                                              config.queryToExecute, config.oracleServerHostname, out));
+        }
+        Path file = new Path(config.outputPath);
+        FileSystem fs = FileSystem.get(file.toUri(), new Configuration());
+        try (FSDataOutputStream outStream = fs.create(file);
+             BufferedWriter br = new BufferedWriter(new OutputStreamWriter(outStream, "UTF-8"))) {
+          br.write(out.replaceAll("(?m)^[\\s&&[^\\n]]+|[\\s+&&[^\\n]]+$", "")); //Removes multiline trailing spaces
+        }
+      }
+    } finally {
+      connection.close();
+    }
+  }
+
+  private String buildOracleExportCommand() {
+    String dbConnectionString = config.dbUsername + "/" + config.dbPassword + "@" + config.oracleSID;
+    String colSeparator = getColSeparator(config.format);
+    StringBuilder scriptContent = new StringBuilder();
+    scriptContent.append("set colsep " + "\"" + colSeparator + "\"" + "\n");
+
+    // 32767 is  the maximum limit a linesixe is allowed.
+    scriptContent.append("set linesize 32767" + "\n");
+    scriptContent.append("set newpage none" + "\n");
+    scriptContent.append("set wrap off" + "\n");
+
+    // Set 0 to prevent printing of headings and avoid page breaks
+    scriptContent.append("set pagesize 0" + "\n");
+
+    scriptContent.append("set heading off" + "\n");
+    scriptContent.append("spool on" + "\n");
+    scriptContent.append(config.queryToExecute + "\n");
+    scriptContent.append("spool off" + "\n");
+    scriptContent.append("exit");
+    String tmpSqlScriptFile = config.tmpSQLScriptDirectory + "/tmpHydrator.sql";
+    //Set of commands to be executed in a session
+    String setPath = "export ORACLE_HOME=" + config.oracleHome + "; export ORACLE_SID=" + config.oracleSID + ";";
+    String tmpScriptCreationCommand = "echo '" + scriptContent.toString() + "' > " + tmpSqlScriptFile + ";";
+    String sqlPlusSpoolExecutionCommand = config.oracleHome + "/bin" + "/sqlplus -s " + dbConnectionString +
+      " @" + tmpSqlScriptFile + " |  sed 's/\\s*" + colSeparator + "\\s*/" + colSeparator + "/g';";
+    String tmpSpoolScriptRemovalCommand = "rm " + tmpSqlScriptFile + "";
+    return setPath + tmpScriptCreationCommand + sqlPlusSpoolExecutionCommand + tmpSpoolScriptRemovalCommand;
+  }
+
+  private String getColSeparator(String format) {
+    String columnSeparator = new String();
```

Comment:
shouldn't initialize this. Should just be ``String columnSeparator;``. Then in the ``switch`` case, have a ``default`` case that just has ``throw new IllegalArgumentException(String.format("Unexpected value %s for format. Expected one of %s.", format, EnumSet.of(SeparatorFormat.class)))``